src/controllers/auth.controller.js

```diff
@@ -0,0 +1,46 @@
+
+const jwt = require('jsonwebtoken');
+
+const UserModel = require('../models/user.model');
+const CONSTANTS = require('../lib/constants');
+
+
+const login = async (req, res) => {
+  try {
+    const user = await UserModel.findOne({ email: req.body.email });
+
+    // user not found
+    if (!user) {
+      res.json({
+        success: false,
+        message: 'Authentication failed. User not found.',
+      });
+    }
+
+    // check if correct password
+    if (user.password !== req.body.password) {
+      res.json({
+        success: false,
+        message: 'Authentication failed. Wrong password.',
+      });
+    }
+
+    // correct password, create token
+    const payload = { dummy: 'info' };
+    const token = jwt.sign(payload, CONSTANTS.JWT_ENCRYPTION, {
+      expiresIn: CONSTANTS.JWT_EXPIRATION, // expires in 12 hours
+    });
+
+    res.json({
+      success: true,
+      message: 'User Authenticated',
+      token,
+    });
+  } catch (e) {
+    throw new Error(e);
```

Comment:
The e in `catch(e)` is already an instance of `Error` class. And the default behaviour is to throw an error. So you don't need to catch it and rethrow again.